src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    flags: gtk_sys::GtkCellRendererState,
+    cellarea: *const gdk_sys::GdkRectangle,
+    alignedptr: *mut gdk_sys::GdkRectangle,
+) where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+
+    let rectangle = imp.get_aligned_area(
+        &wrap,
+        &widget,
+        from_glib(flags),
+        &from_glib_borrow(cellarea),
+    );
+    if !alignedptr.is_null() {
+        *alignedptr = *rectangle.to_glib_none().0;
+    }
+}
+
+unsafe extern "C" fn cell_renderer_render<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkCellRenderer,
+    crptr: *mut cairo_sys::cairo_t,
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    bgptr: *const gdk_sys::GdkRectangle,
+    cellptr: *const gdk_sys::GdkRectangle,
+    flags: gtk_sys::GtkCellRendererState,
+) where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+    let cr: cairo::Context = from_glib_borrow(crptr);
+
+    imp.render(
+        &wrap,
+        &cr,
+        &widget,
+        &from_glib_borrow(bgptr),
+        &from_glib_borrow(cellptr),
+        from_glib(flags),
+    );
+}
+
+unsafe extern "C" fn cell_renderer_activate<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkCellRenderer,
+    evtptr: *mut gdk_sys::GdkEvent,
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    pathptr: *const c_char,
+    bgptr: *const gdk_sys::GdkRectangle,
+    cellptr: *const gdk_sys::GdkRectangle,
+    flags: gtk_sys::GtkCellRendererState,
+) -> glib_sys::gboolean
+where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+    let evt: Option<gdk::Event> = from_glib_none(evtptr);
+
+    imp.activate(
+        &wrap,
+        evt.as_ref().unwrap(),
+        &widget,
+        &GString::from_glib_borrow(pathptr),
+        &from_glib_borrow(bgptr),
+        &from_glib_borrow(cellptr),
+        from_glib(flags),
+    )
+    .to_glib()
+}
+
+unsafe extern "C" fn cell_renderer_start_editing<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkCellRenderer,
+    evtptr: *mut gdk_sys::GdkEvent,
+    wdgtptr: *mut gtk_sys::GtkWidget,
+    pathptr: *const c_char,
+    bgptr: *const gdk_sys::GdkRectangle,
+    cellptr: *const gdk_sys::GdkRectangle,
+    flags: gtk_sys::GtkCellRendererState,
+) -> *mut gtk_sys::GtkCellEditable
+where
+    T: CellRendererImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: CellRenderer = from_glib_borrow(ptr);
+    let widget: Widget = from_glib_borrow(wdgtptr);
+    let evt: Option<gdk::Event> = from_glib_none(evtptr);
```

Comment:
See https://github.com/gtk-rs/gtk/pull/937/files#r365506112

Reply:
Changed to `from_glib_borrow`